src/x/checked/types.go

```diff
@@ -39,26 +39,45 @@ type Ref interface {
 	// NumRef returns the ref count to this entity.
 	NumRef() int
 
-	// Finalize will call the finalizer if any, ref count must be zero.
-	Finalize()
-
-	// Finalizer returns the finalizer if any or nil otherwise.
-	Finalizer() resource.Finalizer
-
-	// SetFinalizer sets the finalizer.
-	SetFinalizer(f resource.Finalizer)
-
 	// DelayFinalizer will delay calling the finalizer on this entity
 	// until the closer returned by the method is called at least once.
 	// This is useful for dependent resources requiring the lifetime of this
 	// entityt to be extended.
 	DelayFinalizer() resource.Closer
 
+	// Finalize will call the finalizer if any, ref count must be zero.
+	Finalize()
+}
+
+// RefWithOnFinalize is a ref that also offers a custom finalizer on top
+// checked ref counting.
+type RefWithOnFinalize interface {
```

Comment:
This `interface` is not actually used anywhere, we could remove it?

Reply:
Sounds good, feel free to remove.